jobs/haproxy/templates/drain.erb

```diff
@@ -39,12 +47,24 @@ haproxy_master_pid=$(pgrep -P "$pid" -x haproxy)
 <%- if p("ha_proxy.enable_health_check_http") -%>
 echo "disable frontend health_check_http_url" | /usr/local/bin/socat stdio unix-connect:${sockfile}
 echo "$(date): triggering grace period for process ${haproxy_master_pid}" >> ${logfile}
-sleep <%= p("ha_proxy.drain_frontend_grace_time") -%>
+sleep <%= p("ha_proxy.drain_frontend_grace_time") %>
 <%- end -%>
 
+touch ${lockfile}
 kill -USR1 "${haproxy_master_pid}"
 echo "$(date): triggering drain for process ${haproxy_master_pid}" >> ${logfile}
 
-echo <%= p("ha_proxy.drain_timeout") -%>
+drain_time=0
+drain_timeout=<%= p("ha_proxy.drain_timeout") %>
+while kill -0 "${haproxy_master_pid}" 2>/dev/null; do
+  sleep 1;
+  drain_time=$((drain_time + 1))
+  if [ $drain_time -eq $drain_timeout ]; then
```

Comment:
I would recommend >= to avoid any possible "skipping" issues if this code were ever to be modified incorrectly.
```suggestion
  if [ $drain_time -ge $drain_timeout ]; then
```

Reply:
agreed.